### ISSUE TITLE
error loading module 'socket.core'

### ISSUE THREAD

**Issue by niufwshd** [Kong/kong#3023]
when I followed the install guide to install  kong, and almost the last step,I got such an error:
[root@localhost kong-master]# **kong migrations up -c /etc/kong/kong.conf**
ERROR: /usr/local/share/lua/5.1/luarocks/loader.lua:147: error loading module 'socket.core' from file '/usr/local/lib/lua/5.1/socket/core.so':
	/usr/local/lib/lua/5.1/socket/core.so: wrong ELF class: ELFCLASS64
stack traceback:
	/usr/local/share/lua/5.1/luarocks/loader.lua:147: in function </usr/local/share/lua/5.1/luarocks/loader.lua:144>
	[C]: in function 'require'
	/usr/local/share/lua/5.1/socket.lua:12: in main chunk
	[C]: in function 'require'
	./kong/core/globalpatches.lua:63: in function <./kong/core/globalpatches.lua:3>
	./kong/cmd/init.lua:1: in main chunk
	[C]: in function 'require'
	/usr/local/bin/kong:7: in function 'file_gen'
	init_worker_by_lua:47: in function <init_worker_by_lua:45>
	[C]: in function 'xpcall'
	init_worker_by_lua:54: in function <init_worker_by_lua:52>
 this repo is great and complex for me, I have no idea about this,please help, thank a lot!

**Comment by bungle** [comment]
For some reason it seems like you have wrong LuaSocket:
https://luarocks.org/modules/luarocks/luasocket

In your system, or it was compiled with wrong settings. 32b vs. 64b or something.

How did you install that?

**Comment by hishamhm** [comment]
Yes, this is definitely a 32-bit vs 64-bit issue. Your LuaRocks installation was compiled to build 64-bit libraries, but your `resty` interpreter was compiled as a 32-bit library.

**Comment by niufwshd** [comment]
@bungle @hishamhm ,Yes,I think you are right! I  rememered I download one luasocket source package and install it,I didn't set neither of 32b or 64b.   When I executed "kong migration up -c /etc/kong/kong.conf", the luasocket's dir is /usr/local/lib/lua/5.1/socket", follow bungle's reply, I have installed luasocket 3.0rc1-2  in "/usr/local/luarocks" directory, then I  executed above kong command,the error was same. two luasocket directories are diffrent. so , I need to  change any kong setting?, but I realy agree the 32bit/64bit solution。but I don't know the detail steps, I didn't find any lua/luarocks optin about it. can you give any advice ?  my redhat is 32bit.

**Comment by niufwshd** [comment]
@@bungle @hishamhm 
 Today I  reinstall my  rhel6.4 system and kong by source method. When I executed " luarocks install kong 0.11.1-0",but i couldn't find any kong config file like "kong.conf" or "kong.conf.default". I searched this link https://github.com/Kong/kong/issues/1548, I think it's the solution. but I don't know the detail steps,can you tell me? thank you first!

**Comment by kikito** [comment]
Hi @niufwshd,

Did you find the source of your problem?
